### PR TITLE
[alignment] add KimiK2_EP2 precision alignment case

### PR DESCRIPTION
**Pull request by liym27**
## What

在 `scripts/alignment_model_accuracy/` 新增 Kimi-K2 精度对齐监控用例 `KimiK2_EP2/`，配置约定对标同目录 `GLM45Air_EP2`（#1903）。

- 并行配置：单机 **2 卡，TP=1 / EP=2 / PP=1**
- 精度：bf16，`seed=23`，单 step（`max_steps=1` / `--train_iters 1`）
- 判定：`compare_loss.py` 逐 step 比对两侧 `per_token_loss` / `final_loss` 的 md5

> 说明：本 PR 早期版本为 `KimiK2_TP8EP8`（单机 8 卡 TP=8/EP=8），因测试环境只有 2 卡，已改为 `KimiK2_EP2`。
> 本次同时 rebase 到最新 develop，解决了 `run_alignment_test.sh` 与 `GLM45Air_EP2` 的登记位置冲突，历史收敛为单 commit。

## Files

| 文件 | 说明 |
|---|---|
| `KimiK2_EP2/KimiK2.yaml` | paddle 侧训练配置（TP=1/EP=2/PP=1，SFT，`use_accuracy_compatible: true`，单 step）|
| `KimiK2_EP2/run_paddle_kimik2.sh` | paddle 侧启动：清分布式 env + 对齐 flags + `paddleformers-cli train` |
| `KimiK2_EP2/run_torch_kimik2.sh` | torch 侧启动：对齐 flags + `megatron sft <ARGS>` |
| `KimiK2_EP2/README.md` | 用例目标、判定口径、两侧配置对应表、端口分配、资产依赖 |
| `run_alignment_test.sh` | 在 `CASES` 中登记本用例（当前为注释状态，见下） |

## 两侧配置对齐

为让严格 md5 可比，以下项两侧成对设置，改动任一侧都需同步另一侧：

| 项 | paddle (`KimiK2.yaml`) | torch (`run_torch_kimik2.sh`) |
|---|---|---|
| attention | `_attn_implementation: eager` | `--attention_backend unfused` + `NVTE_FLASH_ATTN=0` / `NVTE_FUSED_ATTN=0` |
| template | `use_template: false` | `--template dummy --template_backend swift` |
| 截断 | `truncation_strategy: right` | `--truncation_strategy right` |
| padding_free | 未设（默认 false） | `--padding_free False` |
| 样本顺序 | `mix_strategy: concat` / `random_shuffle: false` / `dataloader_shuffle: false` | `--dataset_shuffle False --train_dataloader_shuffle False` |
| MoE aux loss | `router_aux_loss_coef: 0.0` | `--moe_aux_loss_coeff 0` |
| MoE fusion | `moe_expert_fusion: false` / `moe_shared_expert_overlap: false` | `--moe_grouped_gemm False --moe_permute_fusion False` |
| dispatcher | `moe_token_dispatcher_type: alltoall` | `--moe_token_dispatcher_type alltoall` |
| rope | `apply_rope_fusion: false` | 未开 rope fusion |
| 梯度累加 | `amp_master_grad: true` | `--accumulate_allreduce_grads_in_fp32 True` |
| 梯度裁剪 | 默认 `max_grad_norm: 1.0` | `--clip_grad 1.0` |
| 优化器 | 不开 offload / 不开 sharding | `--use_precision_aware_optimizer False --use_distributed_optimizer False` |

资产路径两侧统一到 `/home/.cache/PaddleFormers/Kimi-K2-bf16_2EP/`，保留 `KIMIK2_MODEL` / `KIMIK2_DATASET` 环境变量覆盖口，不写死任何个人共享盘路径。

## 其他约定

- **端口**：paddle 29505 / torch 29504，与 `MinimaxV2.5_EP2`（29501/29500）、`GLM45Air_EP2`（29503/29502）错开。
- **逐层 md5 hook 默认关闭**（`${ENABLE_SAVE_HOOK:-0}` 等），排查首个 diff 层时按需打开，与 `GLM45Air_EP2` 约定一致。
- 两个新增 `.sh` 已补齐 Apache 2.0 版权头，本地 `python3 ci/hooks/copyright.py` 返回 0 且未回写文件。

## `CASES` 当前为注释状态

```bash
CASES=(
    "MinimaxV2.5_EP2 ..."
    "GLM45Air_EP2 ..."
    # "KimiK2_EP2 ./KimiK2_EP2/run_paddle_kimik2.sh ./KimiK2_EP2/run_torch_kimik2.sh"
)
```

`run_case` 任一用例 FAIL 即整体 `exit 1`。2 卡 EP2 配置的 step1 md5 基线尚未 smoke 验证（既往 bit-exact 结论来自 8 卡 TP8/EP8 复现），为避免未验证用例阻塞整条 alignment CI，先以注释形式登记，待 smoke 通过确立基线后单独放开。

## 验证情况

- `bash -n` 三个脚本语法通过
- `yaml.safe_load` 解析 `KimiK2.yaml` 通过
- `ci/hooks/copyright.py` 通过且无回写
- `git rebase origin/develop` 无冲突
- **未做端到端 smoke**（需 2 卡 GPU 与 Kimi-K2 资产），README 已标注该未验证项

### CODE REVIEW

File: scripts/alignment_model_accuracy/KimiK2_EP2/run_paddle_kimik2.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
```
Comment (risemeup1111) [inline scripts/alignment_model_accuracy/KimiK2_EP2/run_paddle_kimik2.sh:1]: P2

新增的两个 `run_*_kimik2.sh` 仍缺少仓库要求的 Apache 2.0 版权头；`ci/hooks/copyright.py` 明确会检查 `.sh`，当前 `Codestyle Check` 也因此在 `copyright_checker` 阶段失败并回写文件。请像同目录的 Minimax 脚本一样补齐版权声明后再提交。

File: scripts/alignment_model_accuracy/KimiK2_EP2/run_torch_kimik2.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    ### model
+    --model_type deepseek_v3
+    --model "${KIMIK2_MODEL}"
+
+    ### data
+    --dataset "${KIMIK2_DATASET}"
+    --columns '{"src": "query", "tgt": "response"}'
+    --max_length 8192
+    --packing False
+    --truncation_strategy delete
```
Comment (risemeup1111) [inline scripts/alignment_model_accuracy/KimiK2_EP2/run_torch_kimik2.sh]: P2

两侧数据预处理仍不是同一语义：`KimiK2.yaml` 使用 `truncation_strategy: right` 和 `padding_free: true`，Torch 这里却使用 `delete` 且没有显式设置 `padding_free`。当样本超过 8192 时，Torch 会删除样本而 Paddle 会截断；即使当前单样本未触发，依赖默认值也会让 md5 对齐结果不可复现。请显式采用与 Paddle 等价的截断/无 padding 设置，并用该 EP2 用例 smoke 验证。

File: scripts/alignment_model_accuracy/KimiK2_EP2/KimiK2.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+### data
+train_dataset_type: erniekit
+eval_dataset_type: erniekit
+train_dataset_path: /root/.cache/PaddleFormers/alignment_paddle.jsonl
```
Comment (risemeup1111) [inline scripts/alignment_model_accuracy/KimiK2_EP2/KimiK2.yaml]: P1

Paddle 侧默认从 `/root/.cache/PaddleFormers/...` 读取模型和数据，但 `model-accuracy-alignment-test` 工作流的容器只把共享 action cache 挂载到 `/home/.cache`，现有 Minimax 用例也使用 `/home/.cache/PaddleFormers`。本 PR 已将该 case 注册进统一 `CASES`，CI 会直接启动它；README 只描述了运行方需要 stage，没有在 workflow 中完成 staging 或导出路径，因此 Paddle 侧会在训练前因资产不存在而失败。请让 PF 默认路径与实际挂载一致，或增加 PF 资产环境变量并在 workflow 中显式 stage/导出。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
| 序号 | 位置 | 优先级 | 规则来源 | 状态 |
| --- | --- | --- | --- | --- |
| 1 | [PF 资产缓存路径](https://github.com/PaddlePaddle/PaddleFleet/pull/1803#discussion_r3818987610) | P1 | [仓库规则：基础评审规则](https://github.com/PaddlePaddle/PaddleFleet/blob/9378b13b0fae54de9d0d927f01027d633ef4d84e/.agents/skills/ai-review/references/base-rules.md) | 🚧 |
| 2 | [两侧数据预处理](https://github.com/PaddlePaddle/PaddleFleet/pull/1803#discussion_r3818987533) | P2 | [仓库规则：基础评审规则](https://github.com/PaddlePaddle/PaddleFleet/blob/9378b13b0fae54de9d0d927f01027d633ef4d84e/.agents/skills/ai-review/references/base-rules.md) | 🚧 |
| 3 | [运行脚本版权头](https://github.com/PaddlePaddle/PaddleFleet/pull/1803#discussion_r3818987527) | P2 | [仓库规则：基础评审规则](https://github.com/PaddlePaddle/PaddleFleet/blob/9378b13b0fae54de9d0d927f01027d633ef4d84e/.agents/skills/ai-review/references/base-rules.md) | 🚧 |
| 4 | [PR 描述与当前配置](https://github.com/PaddlePaddle/PaddleFleet/pull/1803#pullrequestreview-4979530957) | P3 | [仓库规则：基础评审规则](https://github.com/PaddlePaddle/PaddleFleet/blob/9378b13b0fae54de9d0d927f01027d633ef4d84e/.agents/skills/ai-review/references/base-rules.md) | 🚧 |

P3

PR 描述仍写着 `KimiK2_TP8EP8`、单机 TP=8/EP=8，而当前 head 已改为 `KimiK2_EP2`、TP=1/EP=2；请同步目录、并行配置和验证范围，避免评审与提测按旧配置执行。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.6-sol 默认推理级别, please check the suggestions carefully.</sup>
</div>
